lib/components/AlbumScreen/preset_chip.dart

```diff
@@ -0,0 +1,153 @@
+import 'dart:math';
+import 'package:flutter/material.dart';
+import 'package:finamp/services/queue_service.dart';
+import 'package:get_it/get_it.dart';
+
+const _radius = Radius.circular(4);
+const _borderRadius = BorderRadius.all(_radius);
+const _height = 36.0;
+final _defaultBackgroundColour = Colors.white.withOpacity(0.1);
+final _queueService = GetIt.instance<QueueService>();
+
+class PresetChips extends StatefulWidget {
+  const PresetChips({
+    Key? key,
+    required this.type,
+    required this.values,
+    required this.activeValue,
+    this.onTap,
+    this.mainColour,
+    this.onPressed,
+  }) : super(key: key);
+
+  // for future preset types other than "speed"
+  final String type;
+
+  final List<double> values;
+  final double activeValue;
+  final Function()? onTap;
+  final Color? mainColour; // used for different background colours
+  final Function()? onPressed;
+
+  final chipWidth = 55.0;
+
+  @override
+  State<PresetChips> createState() => _PresetChipsState();
+}
+
+class _PresetChipsState extends State<PresetChips> {
+  final _controller = ScrollController();
+  var scrolledAlready = false;
+
+  scrollToActivePreset(double currentValue, double maxWidth) {
+    if (!_controller.hasClients) return false;
+    var offset =
+        (widget.chipWidth + 8.0) * widget.values.indexOf(currentValue) -
+            maxWidth / 2 +
+            widget.chipWidth / 2;
+
+    offset = min(
+        max(0, offset), maxWidth - _controller.position.maxScrollExtent + 15);
+
+    _controller.animateTo(
+      offset,
+      duration: Duration(milliseconds: 350),
+      curve: Curves.easeOutCubic,
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    var nowActiveValue = widget.activeValue;
+
+    return LayoutBuilder(builder: (context, constraints) {
+      return SingleChildScrollView(
+        controller: _controller,
+        scrollDirection: Axis.horizontal,
+        child: Wrap(
+          spacing: 8.0,
+          runSpacing: 8.0,
+          crossAxisAlignment: WrapCrossAlignment.center,
+          children: List.generate(widget.values.length, (index) {
```

Comment:
Maybe move the inner function (`(index) {...}`) into a separate function that is also passed the `Type` and that always returns a `PresetChip`? This way we actually customize the presets based on the type, and because we're using an enum  we can't have the possibility of passing an unknown type.